pallets/automation-time/src/tests.rs

```diff
@@ -461,6 +461,43 @@ fn trigger_tasks_handles_missed_slots() {
 	})
 }
 
+#[test]
+fn trigger_tasks_limits_missed_slots() {
+	new_test_ext(START_BLOCK_TIME).execute_with(|| {
+		let missing_task_id0 = add_task_to_task_queue(ALICE, vec![40], Action::Notify { message: vec![40] });
+		assert_eq!(AutomationTime::get_missed_queue().len(), 0);
+		Timestamp::set_timestamp((SCHEDULED_TIME - 420) * 1_000);
+		schedule_task(ALICE, vec![50], SCHEDULED_TIME - 60, vec![50]);
+		schedule_task(ALICE, vec![60], SCHEDULED_TIME - 120, vec![50]);
+		let missing_task_id3 = schedule_task(ALICE, vec![70], SCHEDULED_TIME - 180, vec![50]);
+		let missing_task_id4 = schedule_task(ALICE, vec![80], SCHEDULED_TIME - 240, vec![50]);
+		let missing_task_id5 = schedule_task(ALICE, vec![90], SCHEDULED_TIME - 300, vec![50]);
+		schedule_task(ALICE, vec![100], SCHEDULED_TIME, vec![50]);
+		Timestamp::set_timestamp(SCHEDULED_TIME * 1_000);
+		LastTimeSlot::<Test>::put((SCHEDULED_TIME - 420, SCHEDULED_TIME - 420));
+		System::reset_events();
+
+		AutomationTime::trigger_tasks(200_000);
+
+		if let Some((updated_last_time_slot, updated_last_missed_slot)) = AutomationTime::get_last_slot() {
```

Comment:
should we not also check that there are tasks scheduled in the times that weren't picked up yet?